File: go/S0005.go
package main

import "fmt"

func longestPalindrome(s string) string {
	if len(s) < 2 {
		return s
	}
	var result string = ""
	for i := 0; i < len(s); i++ {
		for j := i + len(result); j <= len(s); j++ {
			l := 0
			r := len(s[i:j]) - 1
			ok := true
			for true {
				if l >= r {
					break
				}
				if s[i:j][l] != s[i:j][r] {
					ok = false
					break
				} else {
					l++
					r--
				}
			}
			if ok {
				result = s[i:j]
			}
		}
	}
	return result
}

func main() {
    fmt.Println(longestPalindrome("babad"))
}

File: ts/S0005.ts
function longestPalindrome(s: string): string {
    if (s.length < 2) return s;
    let result = '';

    for (let i = 0; i < s.length; i++)
        for (let j = i + result.length; j <= s.length; j++)
            palindRome(s.slice(i, j));
    return result;

    function palindRome(str: string) {
        let l = 0, r = str.length - 1;
        while (l < r) {
            if (str[l] != str[r]) return;
            else l++, r--;
        }
        result = str;
    };
};

console.log(longestPalindrome('babad'))
